### PR TITLE
[0.9.0]: Make database serializable and hashable

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

A breaking change to the database to make it serializable and hashable. This means that the data inserted into the database are now enforced to be primitive or non-primitive built-in types only, except for `sets` and `frozensets`. In essence, the data should be `json` serializable. 

Therefore, a lot of unit tests are expected to fail, because the rounds need to be refactored to not store unserializable data to the db, and the corresponding properties in the synchronized data should now convert them back to the desired type.

Making the database hashable, will allow us to use the digest for the Tendermint app hash. This is a better alternative for the app hash because it guarantees that the state is the same for all the agents at the end of each round.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [x] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

**Review by 0xArdi — APPROVED** [review]

File: docs/api/data/contracts/agent_registry/contract.md [hunk 1]
```diff
@@ -1,71 +0,0 @@
-<a id="autonomy.data.contracts.agent_registry.contract"></a>
```
Comment (DavidMinarsch) [inline docs/api/data/contracts/agent_registry/contract.md:1]: Looks like a merge error? Suggest getting latest main merged and running generators again

File: packages/valory/skills/transaction_settlement_abci/test_tools/integration.py [hunk 2]
```diff
@@ -142,7 +142,24 @@ def sign_tx(self) -> None:
             participant_to_signature[address] = SignaturePayload(
                 sender=address,
                 signature=signature_hex,
-            )
+            ).json
+
+        # FIXME: The following loop is a patch. The
```
Comment (DavidMinarsch) [inline packages/valory/skills/transaction_settlement_abci/test_tools/integration.py:147]: Please open an issue for this and reference here and on Trello.

**Comment by DavidMinarsch** [comment]
Please fill the coverage gaps for `abstract_round_abci`: https://github.com/valory-xyz/open-autonomy/actions/runs/4017500998/jobs/6902151065

File: packages/valory/skills/transaction_settlement_abci/test_tools/integration.py (hunk 2 above)
Comment (Adamantios) [reply packages/valory/skills/transaction_settlement_abci/test_tools/integration.py:147]: Opened #1779 and https://trello.com/c/PxjTZMJz.

**Comment by Adamantios** [comment]
> Please fill the coverage gaps for `abstract_round_abci`: https://github.com/valory-xyz/open-autonomy/actions/runs/4017500998/jobs/6902151065

After all, this is not a merge error. Our generators remove these files as they are not relevant anymore. However, our workflow should have detected this, though allows adding the files back ([6278dd3](https://github.com/valory-xyz/open-autonomy/pull/1754/commits/6278dd37faa4f57839f4deb5d49f9c44ce6ffd7d)).

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**